source/connection-string/tests/invalid-options.yml

```diff
@@ -0,0 +1,126 @@
+tests:
+    -
+        description: "Error on unrecognized option keys"
+        uri: "mongodb://example.com/?foo=bar"
+        valid: false
+        warning: false
+        hosts:
+            -
+                type: "hostname"
+                host: "example.com"
+                port: ~
+        auth: ~
+        options: ~
+    -
+        description: "Error on unsupported option values"
+        uri: "mongodb://example.com/?fsync=ifPossible"
+        valid: false
+        warning: false
+        hosts:
+            -
+                type: "hostname"
+                host: "example.com"
+                port: ~
+        auth: ~
+        options: ~
+    -
+        description: "Repeated option keys with the same value"
+        uri: "mongodb://example.com/?replicaSet=test&replicaSet=test"
+        valid: false
+        warning: false
+        hosts:
+            -
+                type: "hostname"
+                host: "example.com"
+                port: ~
+        auth: ~
+        options:
+            replicaset: "test"
+
+    -
+        description: "Repeated option keys with different values"
+        uri: "mongodb://example.com/?replicaSet=test1&replicaSet=test2"
+        valid: false
+        warning: false
+        hosts:
+            -
+                type: "hostname"
+                host: "example.com"
+                port: ~
+        auth: ~
+        options:
+            replicaset: "test"
+    -
+        description: "List option with one empty option specified"
+        uri: "mongodb://example.com/?readPreferenceTags=dc:ny,rack:1&readPreferenceTags=dc:ny&readPreferenceTags="
+        valid: false
+        warning: false
+        hosts:
+            -
+                type: "hostname"
+                host: "example.com"
+                port: ~
+        auth: ~
+        options:
+            replicaset: "test"
+    -
+        description: "List option with an invalid option specified"
+        uri: "mongodb://example.com/?readPreferenceTags=dc:ny,rack:1&readPreferenceTags=dc:ny&readPreferenceTags=foobarbaz"
+        valid: false
+        warning: false
+        hosts:
+            -
+                type: "hostname"
+                host: "example.com"
+                port: ~
+        auth: ~
+        options:
+            replicaset: "test"
+    -
+        description: "Error on empty integer option values"
+        uri: "mongodb://localhost/?maxIdleTimeMS="
+        valid: false
+        warning: false
+        hosts:
+            -
+                type: "hostname"
+                host: "localhost"
+                port: ~
+        auth: ~
+        options: ~
+    -
+        description: "Error on empty non-integer value for integer option"
+        uri: "mongodb://localhost/?maxIdleTimeMS="
```

Comment:
This is the same URI as above. Also, what's an "empty non-integer value"?

Reply:
I've clarified the wording to specify a "non-empty invalid value" instead